assets/js/timetree.js

```diff
@@ -1,187 +1,261 @@
+            // otherwise, connect to preceding century in this branch
+            target = nodes.length - 2;
+          }
+          links.push({
+            source: branchIndex[branchId],
+            target: target,
+            value: 10
+          });
 
-      if (depth < 5) { // || width > 1) {
+        });
+      }
 
-        // figure out how many branches to draw based on the number of requested children
-        // let angle_start = angle - angle_adjust;
-        // let angle_end = angle + angle_adjust;
-        // let subangle = (angle_start - angle_end) / (n_children - 1);
+      // generate links so we can draw as a network graph
+      // each leaf is connected to its branch+century node
+      sortedLeaves.forEach((leaf, index) => {
+        let branchId = leaf.branch + leaf.century;
+        if (branchId in branchIndex) {
+          links.push({
+            source: index,
+            target: branchIndex[branchId],
+            value: 1
+          })
+        }
+      });
 
-        // subset the available angle range; branches go from + to - angle adjust,
-        // so angle adjustment times two divided by number of children minus 1
-        let subangle_adjust = (angle_adjust * 2) / (n_children - 1);
+      TreeGraph({nodes: nodes, links: links, centuries: centuries});
 
-        // start at full negative angle adjustment
-        let current_angle_adjustment = - angle_adjust;
-        for (let i = 0; i < n_children; i++) {
-          // draw a line for each requested sub branch
+   });
 
-          // randomize the number of child branches
-          let num_children = Math.max(2, Math.floor(Math.random() * 7) + 1);
 
-          // drawline(x, y, length * length_ratio, Math.max(2, n_children - 1), angle + current_angle_adjustment, width/2, depth + 1);
-          drawline(x, y, length * length_ratio, num_children, angle + current_angle_adjustment, width/2, depth + 1);
-          // adjust the angle for the next branch
-          current_angle_adjustment += subangle_adjust;
-        }
-      }
+function TreeGraph({nodes, links, centuries}) {
+  let width = 1000;
+  let height = 300;
+
+  let min_x = -width / 2;
+  let min_y = -height / 2;
+
+  let svg =  d3.select("main")
+      .append("svg")
+      .attr("viewBox", [min_x, min_y, width, height]);
+
+  // create a section for the background
+  let background = svg.append("g")
+    .attr("id", "background");
+
+  // create containers for the leaves by century
+  const leafContainerHeight = 45;
+  const leafContainers = background.selectAll("rect")
+      .data(centuries)
+      .join("rect")
+        .attr("id", d => "c" + d)
+        .attr("height", leafContainerHeight)
+        .attr("width", width)
+        .attr("x", min_x)
+        .attr("y", (d, i) => (min_y + i * leafContainerHeight))
+        .attr('fill', "lightgray")
+        .attr('fill-opacity', 0.1)
+        .attr('stroke', 'gray')
+        .attr('stroke-opacity', 0.5)
+        .attr('stroke-dasharray', 2);
```

Comment:
So these are the ghost nodes?

Reply:
These "leaf containers" are rectangular bands to position things roughly by century — I don't love the styling and probably will shift some of it to CSS, but it seemed good enough for an initial version. Also, in the current layout the leaves aren't strictly contained within those rectangles, so we'll need to either tweak the layout or tweak the century display. Hoping to hold off on that until Gissoo can review and advise.